atomisticparsers/h5md/parser.py

```diff
@@ -0,0 +1,690 @@
+            else:
+                return [value] * n_frames
+
+        # get the steps based on the positions
+        steps = self._data_parser.get(f'{self._path_group_positions_all}.step')
+        if steps is None:
+            self.logger.warning('No step information available in H5MD file. System information cannot be parsed.')
+            return self._system_info
+        self.trajectory_steps = steps
+
+        # get the rest of the particle quantities
+        values_dict = {'system': {}, 'calculation': {}}
+        times = self._data_parser.get(f'{self._path_group_positions_all}.time')
+        values_dict['system']['time'] = times
+        values_dict['calculation']['time'] = times
+        values_dict['system']['positions'] = positions
+        values_dict['system']['n_atoms'] = self._n_atoms
+        system_keys = {'labels': 'system', 'velocities': 'system', 'forces': 'calculation'}
+        for key, sec_key in system_keys.items():
+            path = f'{self._path_group_particles_all}.{self._nomad_to_particles_group_map[key]}'
+            value = self._data_parser.get(path)
+            values_dict[sec_key][key] = get_value(value, steps, path=path)
+
+        # get the box quantities
+        box = self._data_parser.get(f'{self._path_group_particles_all}.box')
+        if box is not None:
+            box_attributes = {'dimension': 'system', 'periodic': 'system'}
+            for box_key, sec_key in box_attributes.items():
+                path = f'{self._path_group_particles_all}.box.{self._nomad_to_box_group_map[box_key]}'
+                value = self._data_parser.get(path, isattr=True)
+                values_dict[sec_key][box_key] = [value] * n_frames if value is not None else None
+            box_keys = {'lattice_vectors': 'system'}
+            for box_key, sec_key in box_keys.items():
+                path = f'{self._path_group_particles_all}.box.{self._nomad_to_box_group_map[box_key]}'
+                value = self._data_parser.get(path)
+                values_dict[sec_key][box_key] = get_value(value, steps, path=path)
+
+        # populate the dictionary
+        for i_step, step in enumerate(steps):
+            self._system_info['system'][step] = {key: val[i_step] for key,val in values_dict['system'].items()}
+            self._system_info['calculation'][step] = {key: val[i_step] for key,val in values_dict['calculation'].items()}
+
+    def parse_observable_info(self):
+        self._observable_info = {
+            'configurational': {},
+            'ensemble_average': {},
+            'correlation_function': {}
+        }
+        thermodynamics_steps = []
+        observables_group = self._data_parser.get('observables')
+        if observables_group is None:
+            return self._observable_info
+
+        def get_observable_paths(observable_group: Dict, current_path: str) -> List:
+            paths = []
+            for obs_key in observable_group.keys():
+                path = obs_key + '.'
```

Comment:
please change to f-formatting for all instances

Reply:
done.